### PR TITLE
[RHCLOUD-38719] Populate null values for resource definitions

### PR DESCRIPTION
**Pull request by astrozzc**
## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-38719

## Description of Intent of Change(s)
HBI team says there are some other services need to make the work around. It would be better for us to fix it on our side.

## Local Testing
Unit tests

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

## Summary by Sourcery

Implement a mechanism to backfill null values in resource definitions for workspaces, specifically handling ungrouped hosts scenarios

New Features:
- Add functionality to replace null workspace values with ungrouped hosts workspace ID

Enhancements:
- Create a utility function to get or create ungrouped workspace for a tenant
- Modify resource definition handling to automatically populate null workspace values

Tests:
- Add comprehensive test cases for backfilling null workspace values
- Test scenarios with existing and non-existing ungrouped hosts workspaces

### CODE REVIEW

File: rbac/internal/utils.py [hunk 1]
```diff
@@ -91,3 +93,31 @@ def delete_bindings(bindings):
             bindings.delete()
         info["relations"] = [stringify_spicedb_relationship(relation) for relation in relations_to_remove]
     return info
+
+
+def get_or_create_ungrouped_workspace(tenant):
```
Comment (lpichler) [inline rbac/internal/utils.py:98]: do we have this logic already somewhere ? can it be  reused ? (in internal view retrieve_ungrouped_workspace)

**Comment by astrozzc** [comment]
This is replaced by: https://issues.redhat.com/browse/RHCLOUD-39981
